src/components/Carousel.tsx

```diff
@@ -1,18 +1,110 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+interface Props {
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite: boolean;
+}
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const [transform, setTransform] = useState(0);
+  const handleNext = () => {
+    let newTransform = transform + step * itemWidth;
+
+    if (images.length - (transform / itemWidth + frameSize) < step) {
+      newTransform = (images.length * itemWidth) - (frameSize * itemWidth);
+    }
+
+    if (transform / itemWidth + frameSize >= images.length && infinite) {
+      newTransform = 0;
+    }
+
+    setTransform(newTransform);
+  };
+
+  const handlePrev = () => {
+    let newTransform = transform - step * itemWidth;
+
+    if ((transform / itemWidth - frameSize) < 0) {
+      newTransform = 0;
+    }
+
+    if (transform < 1 && infinite) {
+      newTransform = (images.length - frameSize) * itemWidth;
+    }
+
+    setTransform(newTransform);
+  };
+
+  useEffect(() => {
+    const wrapElement = document.querySelector('.Carousel') as HTMLElement;
+
+    wrapElement.style.width = `${itemWidth * frameSize}px`;
+    wrapElement.style.transition = `all ${animationDuration}ms ease-in-out`;
+  }, [step, frameSize, itemWidth, animationDuration]);
+
+  return (
+    <div className="MainWrap">
+      <div
+        className="Carousel"
+        style={{ width: `${itemWidth * frameSize}px` }}
+      >
```

Comment:
Do not mutate dom without react. calculate required values before retrun JSX
```suggestion
 
  const width = `${itemWidth * frameSize}px`
  const transition = `all ${animationDuration}ms ease-in-out`

  return (
    <div className="MainWrap">
      <div
        className="Carousel"
        style={{ width, transition }}
      >
```